atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleOverflowWriteTable.java

```diff
@@ -216,6 +219,60 @@ public void delete(List<Entry<Cell, Long>> entries) {
                 args);
     }
 
+    @Override
+    public void delete(RangeRequest range) {
+        String shortTableName = getShortTableName();
+
+        switch (config.overflowMigrationState()) {
+            case UNSTARTED:
+                deleteOverflowRange(config.singleOverflowTable(), shortTableName, range);
+                break;
+            case IN_PROGRESS:
+                deleteOverflowRange(config.singleOverflowTable(), shortTableName, range);
+                deleteOverflowRange(getShortOverflowTableName(), shortTableName, range);
+                break;
+            case FINISHING:
+            case FINISHED:
+                deleteOverflowRange(getShortOverflowTableName(), shortTableName, range);
+                break;
+            default:
+                throw new EnumConstantNotPresentException(
+                        OverflowMigrationState.class, config.overflowMigrationState().name());
+        }
+
+        // delete from main table
+        StringBuilder query = new StringBuilder();
+        query.append(" /* DELETE_RANGE (").append(shortTableName).append(") */ ");
+        query.append(" DELETE /*+ INDEX(m pk_").append(shortTableName).append(") */ ");
+        query.append(" FROM ").append(shortTableName).append(" m ");
+
+        // add where clauses
+        byte[] start = range.getStartInclusive();
```

Comment:
Looks like dupe code, can we expose this and reuse it from the AbstractWriteTable